app/src/main/java/com/osfans/trime/ime/core/TrimeInputMethodService.kt

```diff
@@ -243,7 +243,13 @@ open class TrimeInputMethodService : LifecycleInputMethodService() {
                                 handleReturnKey()
                                 return
                             }
-                            sendDownKeyEvent(eventTime, keyCode, it.modifiers.metaState)
+
+                            if (keyCode in KeyEvent.KEYCODE_NUMPAD_0..KeyEvent.KEYCODE_NUMPAD_EQUALS) {
+                                // ignore KP_X keys, which is handled in `CommonKeyboardActionListener`.
```

Comment:
This may cause IMS cannot reponse the corresponding keyevent from physical keyboard, maybe we should tell if the keyevent is from virtual keyboard or physical keyboard. But now, let us workaround for virtual keyboard first.